initPTB.m

```diff
@@ -152,3 +120,122 @@
 
 
 end
+
+
+function initDebug(cfg)
+
+% init PTB with different options in concordance to the debug Parameters
+Screen('Preference','SkipSyncTests', 0);
+if cfg.debug
+    
+    Screen('Preference', 'SkipSyncTests', 2);
+    Screen('Preference', 'Verbosity', 0);
+    Screen('Preferences', 'SuppressAllWarnings', 2);
+    
+    fprintf('\n\n\n\n')
+    fprintf('########################################\n')
+    fprintf('##   DEBUG MODE. TIMING WILL BE OFF.  ##\n')
+    fprintf('########################################')
+    fprintf('\n\n\n\n')
+    
+    testKeyboards(cfg)
+    
+end
+
+if cfg.testingTranspScreen
+    PsychDebugWindowConfiguration
+end
+
+end
+
+function initKeyboard
+
+% Make sure keyboard mapping is the same on all supported operating systems
+% Apple MacOS/X, MS-Windows and GNU/Linux:
+KbName('UnifyKeyNames');
+
+% Don't echo keypresses to Matlab window
+ListenChar(-1);
+
+end
+
+function cfg = initAudio(cfg)
+    
+    if cfg.initAudio
+        
+        requestedLatency = 3;
+        
+        InitializePsychSound(1);
+
+        cfg.audio.devIdx= [];
+        cfg.audio.playbackMode = 1;
+            
+        if IsWin
+            
+            % get audio device list
+            audioDev = PsychPortAudio('GetDevices');
+            
+            % find output device using WASAPI driver
+            idx = find(...
+                audioDev.NrInputChannels == 0 && ...
+                audioDev.NrOutputChannels == 2 && ...
+                ~cellfun(@isempty, regexp({audioDev.HostAudioAPIName}, 'WASAPI')));
+            
+            % save device ID
+            cfg.audio.devIdx = audioDev(idx).DeviceIndex;
+            
+            % get device's sampling rate
+            cfg.audio.fs = audioDev(idx).DefaultSampleRate;
+
+        end
+        
+        cfg.audio.pahandle = PsychPortAudio('Open', ...
+            cfg.audio.devIdx, ...
+            cfg.audio.playbackMode, ...
+            requestedLatency, ...
+            cfg.audio.fs, ...
+            cfg.audio.channels);
+        
+        % set initial PTB volume for safety (participants can adjust this manually
+        % at the begining of the experiment)
+        PsychPortAudio('Volume', cfg.audio.pahandle, cfg.audio.initVolume);
+        
+        cfg.audio.pushSize  = cfg.fs*0.010; %! push N ms only
```

Comment:
Not sure if we want to initialize those for all audio experiment.

Plus there is some hard coding there I am not comfortable with.

Reply:
that's a good example to practice. so when one experiment is more specific, how shall we proceed? change things in setDefaults.m?
I do not even use these two lines but my colleague's script which calls the same initPTB.m needs these two lines.

Reply:
> that's a good example to practice. so when one experiment is more specific, how shall we proceed? change things in setDefaults.m?
> I do not even use these two lines but my colleague's script which calls the same initPTB.m needs these two lines.

I think I would keep the most general stuff in the setDefaults and initPTB. Those we know will be needed in most use case.

Then either each user updates cfg after calliing initPTB like this

```
[cfg] = initPTB(cfg)
cfg.audio.pushSize  = cfg.audio.fs * 0.010;
```

Or if needed we can create additional "flags" for some specific configuration: I am thinking of doing this for issue #38

Reply:
will merge for now and open an issue for this.